kong/plugins/opentelemetry/otlp.lua

```diff
@@ -72,9 +77,39 @@ local function transform_events(events)
   return pb_events
 end
 
+-- this function is to translate the span to otlp span, but of universal span format
+local function translate_span(span)
+  local trace_id = span.trace_id
+  local len = #trace_id
+  local new_id = trace_id
+
+  if len == TRACE_ID_LEN then
+    return span
+  end
+
+  -- make sure the trace id is of 16 bytes
+  if len > TRACE_ID_LEN then
+    new_id = trace_id:sub(-TRACE_ID_LEN)
+
+  elseif len < TRACE_ID_LEN then
+    new_id = NULL:rep(TRACE_ID_LEN - len) .. trace_id
+  end
+
+  local translated = shallow_copy(span)
```

Comment:
Why is a `shallow_copy` needed?

I would have expected this function to do something like:

```
local function adjust_trace_id_for_oltp(trace_id)
```

And the usage would be, when creating the root span:
```
root_span.trace_id = adjust_trace_id_for_oltp(trace_id)
```

And inside `transform_span`:
```
local pb_span = {
    trace_id = adjust_trace_id_for_oltp(span.trace_id),
```

Reply:
Also, I don't recommend using abstract names (like `transform_span` or `translate_span`) when the function is only doing one modification on one field. The tradeoff ("making the code easier to modify in the future if we find that we need to do more modifications in the future") is not worth the upfront cost ("making the code more difficult to read now"). Future us can always group several modifications into one if the need arises.

Reply:
> Why is a `shallow_copy` needed?

The span passed in may also be used by other plugins. We need to preserve as much original information as possible, and only do the transformation for exporting/propagating.

Reply:
I agree naming is a problem, but `transform_span` is used as some kind of interface for tracing plugins (we can see a function with the same name from datadog-tracing).
I'm adding comments and renaming the internal function.

Reply:
And the function is in the file "otlp.lua" so I think we do not need to suffix it with "\_otlp".

Reply:
> And the usage would be, when creating the root span:

We cannot just modify the `root_span` given the reason: https://github.com/Kong/kong/pull/10332/files#r1115411177